File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun maxLevelSum(root: TreeNode?): Int {
        var maxSum = Integer.MIN_VALUE; var maxLevel = 1
        var currLevel = 0
        val q: Queue<TreeNode> = LinkedList<TreeNode>()
        q.add(root!!)
        while (q.isNotEmpty()) {
            val size = q.size
            var levelSum = 0
            currLevel++
            for (i in 0 until size) {
                val node = q.remove()
                levelSum += node.`val`
                node.left?.let {q.add(it)}
                node.right?.let {q.add(it)}
            }
            if (maxSum < levelSum) {
                maxSum = levelSum
                maxLevel = currLevel
            }
        }
        return maxLevel
    }
}

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        q = []
        q.append(root)
        maxSum = -10**5 -1
        maxLevel = 1
        level = 0
        while q:
            level += 1
            levelSum = 0
            size = len(q)
            for i in range(size):
                node = q.pop(0)
                levelSum += node.val
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            if maxSum < levelSum:
                maxSum = levelSum
                maxLevel = level
        return maxLevel
            
        